execution/function/operator.go

```diff
@@ -285,10 +292,62 @@ func (o *functionOperator) loadSeries(ctx context.Context) error {
 		}
 
 		o.series = make([]labels.Labels, len(series))
+
+		var labelJoinDst string
+		var labelJoinSep string
+		var labelJoinSrcLabels []string
+		if o.funcExpr.Func.Name == "label_join" {
+			l := len(o.funcExpr.Args)
+			if l < 3 {
+				err = errors.New("invalid label_join function arguments: atleast 3 arguments required")
+				return
+			}
+
+			labelJoinDstLiteral, ok := o.funcExpr.Args[1].(*parser.StringLiteral)
+			if !ok {
+				err = errors.New("invalid label_join function arguments: dst label is not a string")
```

Comment:
can you write a test with those error scenarios?

Reply:
Yeah, of course I can! Generally, queries like `label_join(foo{}, "label", http_total_requests, 23)`  (i.e. if the arguments are not strings) should trigger these errors.
I just need to create a dedicated `TestLabelJoin` function for that. Because `TestInstantQuery` doesn't check for error string equality.

Reply:
Seems like these type of invalid cases are handled by parser already. So these checks shouldn't be triggered here. Should I remove these checks?

Reply:
Oh i think `TestQueriesAgainstOldEngine` tests also the error messages so that would be a quick way to check. If that is already handled by the parser and we have tests in case this should change then removing the checks seems like a better option!

Reply:
Yeah, I tested one invalid function `label_join(http_requests_total{}, "label", http_requests{series="3"}, 12)` and the returned error was `unexpected error: 1:44: parse error: expected type string in call to function "label_join", got instant vector` which is coming from the parser.